### PR TITLE
Boolean matcher

### PR DESCRIPTION
**Pull request by HJewkes**

### CODE REVIEW

File: client/src/main/java/io/split/engine/matchers/Transformers.java [hunk 1]
```diff
@@ -58,6 +60,24 @@ public static Long asDateHourMinute(Object obj) {
...
+        if (obj == null) {
+            return null;
+        }
+
+        if (obj instanceof Boolean) {
+            return (Boolean) obj;
+        }
+
+        if (obj instanceof String) {
+            if (Sets.newHashSet("true", "false").contains(((String) obj).toLowerCase())) {
```
Comment (SenhorCastor) [inline client/src/main/java/io/split/engine/matchers/Transformers.java]: have a static set so its not initialized every time?, same with the (String obj)

File: client/src/main/java/io/split/engine/matchers/BooleanMatcher.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public boolean match(Object matchValue, String bucketingKey, Map<String, Object> attributes, SplitClientImpl splitClient) {
+        if (matchValue == null) {
+            return false;
+        }
+
+        Boolean valueAsBoolean = asBoolean(matchValue);
+
+        return valueAsBoolean != null && valueAsBoolean == _booleanValue;
```
Comment (adilaijaz) [inline client/src/main/java/io/split/engine/matchers/BooleanMatcher.java]: valueAsBoolean.booleanValue() == _booleanValue

File: client/src/main/java/io/split/engine/matchers/BooleanMatcher.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        BooleanMatcher that = (BooleanMatcher) o;
+
+        return _booleanValue == that._booleanValue;
```
Comment (adilaijaz) [inline client/src/main/java/io/split/engine/matchers/BooleanMatcher.java]: same here. Wouldn't we want to do a .booleanValue() here.

Comment (HJewkes) [inline client/src/main/java/io/split/engine/matchers/BooleanMatcher.java]: Unboxing for equality checks is unnecessary, Java does it automatically: https://stackoverflow.com/questions/30454635/when-using-for-a-primitive-and-a-boxed-value-is-autoboxing-done-or-is-unbox

File: client/src/test/java/io/split/engine/matchers/AttributeMatcherTest.java [hunk 4]
```diff
@@ -54,6 +54,17 @@ public void works_less_than_or_equal_to() {
     }
 
     @Test
+    public void works_boolean() {
+        AttributeMatcher matcher = new AttributeMatcher("value", new BooleanMatcher(true), false);
+        assertThat(matcher.match("ignore", null, ImmutableMap.<String, Object>of("value", true), null), is(true));
+        assertThat(matcher.match("ignore", null, ImmutableMap.<String, Object>of("value", "true"), null), is(true));
```
Comment (patricioe) [inline client/src/test/java/io/split/engine/matchers/AttributeMatcherTest.java]: could you please add a test with upper case as input?

**Comment by patricioe** [comment]
Approved on my end. Left a comment to add some more tests.

**Pull request merged**